avm/res/aad/domain-service/tests/e2e/waf-aligned/main.test.bicep

```diff
@@ -0,0 +1,154 @@
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-aad-domainservices-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'aaddswaf'
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableTelemetry bool = false
+
+@description('Optional. A token to inject into the name of each resource. This value can be automatically injected by the CI.')
+param namePrefix string = '#_namePrefix_#'
+
+var varDiagSettings = [
+  {
+    name: 'customSetting'
+    eventHubName: diagnosticDependencies.outputs.eventHubNamespaceEventHubName
+    eventHubAuthorizationRuleResourceId: diagnosticDependencies.outputs.eventHubAuthorizationRuleId
+    storageAccountResourceId: diagnosticDependencies.outputs.storageAccountResourceId
+    workspaceResourceId: diagnosticDependencies.outputs.logAnalyticsWorkspaceResourceId
+  }
+]
+var varLock = {
+  kind: 'None'
+  name: 'myCustomLockName'
+}
+var varReplicaSets = [
+  {
+    location: 'NorthEurope'
+    subnetId: nestedDependencies.outputs.subnetResourceId
+  }
+]
+var varTags = {
+  'hidden-title': 'This is visible in the resource name'
+  Environment: 'Non-Prod'
+  Role: 'DeploymentValidation'
+}
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: resourceLocation
+}
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-nestedDependencies'
+  params: {
+    location: resourceLocation
+    virtualNetworkName: 'dep-${namePrefix}-vnet-${serviceShort}'
+    keyVaultName: 'dep-${namePrefix}-kv-${serviceShort}'
+    managedIdentityName: 'dep-${namePrefix}-msi-${serviceShort}'
+    namePrefix: namePrefix
+    certDeploymentScriptName: 'dep-${namePrefix}-ds-${serviceShort}'
+  }
+}
+
+// Diagnostics
+// ===========
+module diagnosticDependencies '../../../../../../utilities/e2e-template-assets/templates/diagnostic.dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-diagnosticDependencies'
+  params: {
+    storageAccountName: 'dep${namePrefix}diasa${serviceShort}01'
+    logAnalyticsWorkspaceName: 'dep-${namePrefix}-law-${serviceShort}'
+    eventHubNamespaceEventHubName: 'dep-${namePrefix}-evh-${serviceShort}'
+    eventHubNamespaceName: 'dep-${namePrefix}-evhns-${serviceShort}'
+    location: resourceLocation
+  }
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+resource keyVault 'Microsoft.KeyVault/vaults@2022-07-01' existing = {
```

Comment:
I'm a bit blown away - does that work nowadays? I recall, that originally, you could not reference an `existing` resource if set resource did not exist at the start of the deployment (which in this case it definitely doesn't as it is deployed via the dependencies).